lib/react_client/react_interop.dart

```diff
@@ -423,10 +425,26 @@ class ReactDartInteropStatics2 implements ReactDartInteropStatics {
 /// See [ReactDartInteropStatics], [createReactDartComponentClass].
 class ComponentStatics<T extends Component> {
   final ComponentFactory<T> componentFactory;
-
   ComponentStatics(this.componentFactory);
 }
 
+/// An object that stores static methods and information for a specific component class.
+///
+/// This object is made accessible to a component's JS ReactClass config, which
+/// passes it to certain methods in [ReactDartInteropStatics2].
+///
+/// [ComponentStatics2] exists in addition to [ComponentStatics] because
+/// [Component2] brings new static lifecycle methods that are exposted via
+/// `componentInstace`. Because [Component] does not need an instance to pull
+/// static lifecycle methods from, it made sense to create [ComponentStatics2].
+///
+/// See [ReactDartInteropStatics2], [createReactDartComponentClass2].
+class ComponentStatics2<T extends Component2> {
```

Comment:
No need for the generics here I don't think.

Reply:
I believe it may be necessary here because `ComponentFactory` is defined with the `T` generic. Is there a workaround to that?

Reply:
@joebingham-wk you would just do this instead:

```dart
class ComponentStatics2 {
  final ComponentFactory<Component2> componentFactory;
  final Component2 instanceForStaticMethods;
  ComponentStatics2(this.componentFactory, {this.instanceForStaticMethods});
}
```

Reply:
Well, that was simple. I should have seen that - I'm sorry for making you type it! Code is updated with that change.